### PR TITLE
Logger refactor: log message polishing

### PR DESCRIPTION
**Pull request by EmilianoSanchez**
# Javascript commons library

## What did you accomplish?

- Changes:
  - reuse messages for input validation, in favor of size reduction.
  - updated log message to follow the format: '[LOGLEVEL] splitio => optionalPrefixOrCategory: message'
  - replaced 'Factory instantiation' category to 'settings' category for split filter validator logs, like the other settings validators (except API Key validation logs).
  - removed 'Retrieving SDK client.' to reuse 'Retrieving default SDK client.'

## Extra Notes

### CODE REVIEW

**Review by NicoZelaya — APPROVED** [review]

File: src/evaluator/value/index.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { SplitIO } from '../../types';
 import { IMatcherDto } from '../types';
 import { ILogger } from '../../logger/types';
 import sanitizeValue from './sanitize';
-import { DEBUG_24, WARN_1, WARN_0 } from '../../logger/constants';
+import { DEBUG_24, WARN_ENGINE_NO_ATTRIBUTES, WARN_ENGINE_INVALID_VALUE } from '../../logger/constants';
```
[the remark this answers is not in the document]
Comment (EmilianoSanchez) [reply src/evaluator/value/index.ts:5]: done in next commit

File: src/logger/constants.ts [hunk 2]
```diff
@@ -1,151 +1,140 @@
-// commons
```
[the remark this answers is not in the document]
Comment (EmilianoSanchez) [reply src/logger/constants.ts:1]: will be done in a coming commit, when renaming constants

File: src/logger/index.ts [hunk 3]
```diff
@@ -31,36 +32,42 @@ export class Logger implements ILogger {
...
   }
 
-  private _log(level: LogLevel, text: string, args?: any[]) {
+  private _log(level: LogLevel, msg: string | number, args?: any[]) {
     if (this._shouldLog(level)) {
-      if (args) text = sprintf(text, args);
-      const formattedText = this._generateLogMessage(level, text);
+      if (typeof msg === 'number') {
+        const format = this.codes.get(msg);
+        if (format) msg = sprintf(format, args);
```
[the remark this answers is not in the document]
Comment (EmilianoSanchez) [reply src/logger/index.ts:67]: yes! good idea. It was added in next PR

File: src/logger/messages/error.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { ERROR_0, ERROR_2, ERROR_CLIENT_LISTENER, ERROR_4, ERROR_5, ERROR_7, ERROR_9, ERROR_10, ERROR_11, ERROR_12, ERROR_EVENT_TYPE_FORMAT, ERROR_NOT_PLAIN_OBJECT, ERROR_SIZE_EXCEEDED, ERROR_NOT_FINITE, ERROR_CLIENT_DESTROYED, ERROR_NULL, ERROR_TOO_LONG, ERROR_INVALID_KEY_OBJECT, ERROR_INVALID, ERROR_EMPTY, ERROR_EMPTY_ARRAY, ERROR_INVALID_IMPRESSIONS_MODE, ERROR_39, SETTINGS_LB, ENGINE_COMBINER_LB, SYNC_OFFLINE_LB, SYNC_STREAMING_LB, IMPRESSIONS_TRACKER_LB } from '../constants';
+
+export const codesError: [number, string][] = [
+  // evaluator
+  [ERROR_0, ENGINE_COMBINER_LB + 'Invalid Split provided, no valid conditions found'],
```
[the remark this answers is not in the document]
Comment (EmilianoSanchez) [reply src/logger/messages/error.ts:5]: done

**Pull request merged**